### DISCUSSION TITLE
Is that possible to define "SET NULL" for only one column of composite foreign key?

### DISCUSSION THREAD

**Discussion by AlexanderFarkas** [sqlalchemy/sqlalchemy#10727] [Usage Questions]
I believe, Postgres [supports](https://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=d6f96ed94e73052f99a2e545ed17a8b2fdc1fb8a) it.

Also, in the [documentation](https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-FK) (at the very end) 
Wondering, if it's possible to do it in detectable way for SQLAlchemy

**Comment by AlexanderFarkas** [comment]
If I code it like:
```python
ForeignKeyConstraint(..., ondelete="set null(payment_id)")
```

It throws:
```
 Unexpected SQL phrase: 'set null(payment_id)' (matching against '^(?:RESTRICT|CASCADE|SET NULL|NO ACTION|SET DEFAULT)$') ```

**Comment by CaselIT** [comment]
Hi,

Can't you use `ondelete='SET NULL'`?

**Reply by CaselIT** [reply]
oh sorry, reading only the text is was not clear. No I don't think that syntax is supported

**Reply by CaselIT** [reply]
note that there isn't a reason that this is not supported, simply no-one asked for it before.
ForeignKeyConstraint/ForeignKey already support dialect kwargs so it should be pretty straight forward to add. It could be `postgresql_columns` that's supported by both `set null` and `set default`: https://www.postgresql.org/docs/16/sql-createtable.html

Would you like to try creating a PR for it?

**Reply by AlexanderFarkas** [reply]
Yes. Working on it 😄

**Reply by CaselIT** [reply]
If you have questions feel free to ask for directions

**Comment by jonaengs** [comment]
Just encountered this same problem. Would really love to see support for this postgres feature get added. @AlexanderFarkas did end up making a PR? I looked around a bit but couldn't find anything.

**Reply by AlexanderFarkas** [reply]
No, I didn't. My team decided against it and instead implemented schema-based tenancy.

**Comment by CaselIT** [comment]
I've created https://github.com/sqlalchemy/sqlalchemy/issues/11946 to better track it.
